linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/SourceMapWriter.scala

```diff
@@ -169,8 +170,14 @@ object SourceMapWriter {
     }
 
     private def writePendingSegment(): Unit = {
-      if (pendingColumnInGenerated >= 0)
-        doWriteSegment(pendingColumnInGenerated, pendingPos, pendingName)
+      if (pendingColumnInGenerated >= 0) {
+        // Allocate a name string *before* we write a fragment so we can cache it.
```

Comment:
Obsolete with the revert. The idea was that we convert `OriginalName` to `String` before we create a fragment so that when we write a fragment, we already have a `String`, so we're faster.